esp-hal/src/spi/master.rs

```diff
@@ -1191,21 +1137,18 @@ pub mod dma {
 
             self.spi
                 .start_read_bytes_dma(ptr, len, &mut self.channel.rx, false)?;
-            Ok(SpiDmaTransfer {
-                spi_dma: self,
-                buffer,
-            })
+            Ok(SpiDmaTransfer { spi_dma: self })
         }
 
         #[cfg_attr(feature = "place-spi-driver-in-ram", ram)]
-        pub fn write<TXBUF>(
-            mut self,
+        pub fn write<'t, TXBUF>(
+            &'t mut self,
             data_mode: SpiDataMode,
             cmd: Command,
             address: Address,
             dummy: u8,
-            buffer: TXBUF,
-        ) -> Result<SpiDmaTransfer<'d, T, C, TXBUF, M>, super::Error>
+            buffer: &'t mut TXBUF,
```

Comment:
Because this is write, shouldn't this be `&'t TXBUF`?

Reply:
I thought about it but thought `&'t mut` would resemble the move-in / move-out pattern more closely but that was probably a bit extreme. Probably it's fine if there could be immutable borrows for the TX data 🤔 

Thinking about it even more it's also less surprising to the user to have it `&'t`